File: 3005.py
from collections import defaultdict

class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        count = defaultdict(lambda : 0)
        max_count = 0

        for num in nums:
            count[num] += 1
            max_count = max(max_count, count[num])

        res = 0
        for key, value in count.items():
            if(value == max_count):
                res += value

        return res

File: 3005.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> count;
        int max_count = 0, res = 0;

        for(int i = 0; i < nums.size(); i++) {
            count[nums[i]]++;
            max_count = fmax(max_count, count[nums[i]]);
        }
        for(auto it = count.begin(); it != count.end(); it++) {
            if(it->second == max_count) {
                res++;
            }
        }
        return res * max_count;
    }
};